qst/src/main/java/io/deephaven/qst/table/ParentsVisitor.java

```diff
@@ -0,0 +1,236 @@
+package io.deephaven.qst.table;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+/**
+ * A visitor that returns the parent tables (if any) of the given table.
+ */
+public class ParentsVisitor implements Table.Visitor {
+
+    /**
+     * A traversal of the table's parents. Does not perform de-duplication.
+     *
+     * @param table the table
+     * @return the parents stream
+     */
+    public static Stream<Table> getParents(Table table) {
+        return table.walk(new ParentsVisitor()).getOut();
+    }
+
+    /**
+     * Create a depth-first ordered set.
```

Comment:
Define your depth-first ordering more specifically. I would expect pre-order, but you may have another plan. See https://en.wikipedia.org/wiki/Depth-first_search